server/data/applications.ts

```diff
@@ -1,273 +1,284 @@
-     * throws an error otherwise
-     */
-    deleteApplication: async (cycleId: string, appId: string): Promise<boolean> => {
-        if(!cycleId || !checkObjectId(cycleId)) {
-            throw new Error("a proper cycle id must be provided.")
-        }
-
-        if(!appId || !checkObjectId(appId)) {
-            throw new Error("A proper application id must be provided.");
-        }
-
-        //Finds the cycle with cycleId
-        const cycleCollection: any = await cycles();
-        let cycle: CycleObject = cycleCollection.findOne({
-            _id: new ObjectId(cycleId)
-        })
-
-        //Finds position of the application to be deleted
-        let appPos: number = null;
-        for(let x=0; x<cycle["applications"].length; x++) {
-            if(cycle["applications"][x]["_id"].toString() === appId) {
-                appPos = x;
-                break;
-            }
-        }
-        if(appPos === null) {
-            throw new Error("Unable to find an application with that id.");
-        }
-
-        //Slices the cycle application array to remove application with appId
-        cycle["applications"] = cycle["applications"].splice(appPos, 1);
-        let updatedInfo = await cycleCollection.updateOne({_id: new ObjectId(cycleId)}, {$set: cycle});
-        if(updatedInfo.modifiedCount === 0) {
-            throw new Error("Could not update application.");
-        }
-
-        return true;
+    return application;
+};
+
+/**
+ * @description Get an application by id from a given cycle
+ * @param {string} cycleId cycle id
+ * @param {string} applicationId Application id
+ * @returns {Promise<Object>} Cycle Object with given id if found 
+ * and throws an error otherwise
+ */
+ export const getApplicationFromCycleById = async (cycleId: string, applicationId: string): Promise<ApplicationObject> => {
+    if(!cycleId || !checkObjectId(cycleId))
+        throw new Error("A proper cycle id must be provided.");
+
+    if(!applicationId || !checkObjectId(applicationId))
+        throw new Error("A proper application id must be provided.");
+
+    const cyclesCollection = await cycles();
+    const cycle: CycleObject = await cyclesCollection.findOne({ _id: new ObjectId(cycleId) });
+    if(cycle === null)
+        throw new Error("User has no cycles.");
+
+    let application: ApplicationObject = cycle.applications.find(a => a._id.toString() === applicationId);
+    if(application === undefined)
+        throw new Error("No application found with that id.")
+    
+    return application;
+};
+
+/**
+ * @description Finds all applications from a given cycle
+ * @param {string} cycleId The cycle id to get all applications from
+ * @returns {Promise<ApplicationObject[]>} Array of applications belonging to a given cycle,
+ * otherwise throws and error
+ */
+export const getAllApplications = async (cycleId: string): Promise<ApplicationObject[]> => {
```

Comment:
This is fine, but do we want a separate function to get all applications from the current cycle, since we decided it was proper find the current cycle id in the data functions as opposed to the routes?

Reply:
Or, is it because the client will always have access to the current cycle id, so they can make a request with it?